### ISSUE TITLE
rule prefer-named-capture-group should only be effective for ES9+

### ISSUE THREAD

**Issue by erwindon** [eslint/eslint#14959]
**What rule do you want to change?**
prefer-named-capture-group

**Does this change cause the rule to produce more or fewer warnings?**
fewer warnings

**How will the change be implemented? (New option, new default behaviour, etc.)?**
new default behaviour

**Please provide some example code that this change will affect:**
use a replace statement with a numeric group reference in its replace pattern while using ES6 or any other ES version 1..8.
```js
"kjhgghgh".replace(/xxx(yyy)zzz/, "aaa$1bbb")
```

This will raise ```Capture group '(yyy)' should be converted to a named or non-capturing group```, but named groups are not supported in ES6 and a non-capturing group is useless in this context.

The use of ES6 was indicated by setting `eslintConfig.parserOptions.ecmaVersion` to 6 in `package.json`.

**What does the rule currently do for this code?**
The current rule currently suggests to use named capture groups, even while using an ES version where that is not supported.

**What will the rule do after it's changed?**
It will not report this for ES 1..8

**Are you willing to submit a pull request to implement this change?**
no, as I'm not skilled in the inner structure/workings of eslint.

**Comment by mdjermanovic** [comment]
Hi @erwindon, thanks for the issue!

> The use of ES6 was indicated by setting `eslintConfig.parserOptions.ecmaVersion` to 6 in `package.json`.

ESLint can be used with third-party parsers that have different defaults, or even don't have `ecmaVersion` as a parser option, so we aim to avoid using `parserOptions.ecmaVersion` for auto-disabling or altering the logic of rules because it isn't reliable. 

> This will raise Capture group '(yyy)' should be converted to a named or non-capturing group, but named groups are not supported in ES6 and a non-capturing group is useless in this context.

The solution is to disable this rule, as recommended in the [documentation](https://eslint.org/docs/rules/prefer-named-capture-group#when-not-to-use-it):

> If you are targeting ECMAScript 2017 and/or older environments, you can disable this rule, because this ECMAScript feature is only supported in ECMAScript 2018 and/or newer environments.

(we should change the wording from "you can disable" to "you should disable").

I'm closing this issue as we have already decided on not making these kinds of changes (e.g., https://github.com/eslint/eslint/pull/12624 was a similar proposal).
